tf-modules/elasticsearch/iam.tf

```diff
@@ -22,37 +22,18 @@ resource "aws_iam_role_policy_attachment" "data-node-attach-ec2-discovery" {
   policy_arn = "${aws_iam_policy.ec2-discovery-policy.arn}"
 }
 
-module "credstash-reader" {
-  source = "../credstash"
-  name_prefix = "${var.name_prefix}-elasticsearch"
-  create_reader_policy = true
-  kms_key_arn = "${var.credstash_kms_key_arn}"
-  table_name = "${var.credstash_table_name}"
-}
-
-resource "aws_iam_role_policy_attachment" "master-node-credstash-reader-policy-attachment" {
-  count = "${var.master_node_count}"
-  role = "${element(aws_iam_role.master-node-role.*.name, count.index)}"
-  policy_arn = "${module.credstash-reader.reader_policy_arn}"
-}
-
-resource "aws_iam_role_policy_attachment" "data-node-credstash-reader-policy-attachment" {
-  count = "${var.data_node_count}"
-  role = "${element(aws_iam_role.data-node-role.*.name, count.index)}"
-  policy_arn = "${module.credstash-reader.reader_policy_arn}"
+module "credstash-grant" {
+  source            = "../credstash-grant"
+  kms_key_arn       = "${var.credstash_kms_key_arn}"
+  reader_policy_arn = "${var.credstash_reader_policy_arn}"
+  roles_count       = "${var.master_node_count + var.data_node_count}"
+  roles_arns        = ["${concat(aws_iam_role.master-node-role.*.arn, aws_iam_role.data-node-role.*.arn)}"]
+  roles_names       = ["${concat(aws_iam_role.master-node-role.*.name, aws_iam_role.data-node-role.*.name)}"]
```

Comment:
ah, I guess this might be a good example to mention / use in the docs (for multiple roles).